File: app/src/release/java/at/tomtasche/reader/ui/OpenFileIdling.java
package at.tomtasche.reader.ui;

/**
 * Release variant: does nothing. The debug variant backs these calls with an espresso
 * CountingIdlingResource; keeping the espresso dependency out of release builds is the whole point
 * of the split.
 */
public final class OpenFileIdling {

    private OpenFileIdling() {}

    public static void increment() {}

    public static void decrement() {}
}

File: app/src/debug/java/app/opendocument/droid/ui/OpenFileIdling.kt
package app.opendocument.droid.ui

import androidx.test.espresso.IdlingResource
import androidx.test.espresso.idling.CountingIdlingResource

/**
 * Debug variant: a real espresso idling resource, so instrumented tests can wait for a document to
 * finish loading. The release variant does nothing, keeping espresso out of the shipped apk.
 */
object OpenFileIdling {

    private val RESOURCE = CountingIdlingResource("MainActivity.openFileIdlingResource")

    val idlingResource: IdlingResource
        get() = RESOURCE

    fun increment() {
        RESOURCE.increment()
    }

    fun decrement() {
        if (!RESOURCE.isIdleNow) {
            RESOURCE.decrement()
        }
    }
}
